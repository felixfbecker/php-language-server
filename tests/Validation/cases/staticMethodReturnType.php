<?php

class FooClass {
    public static function staticFoo(): FooClass {
        return new FooClass();
    }

    public static function staticSelf(): self { }

    public function bar() { }
}
