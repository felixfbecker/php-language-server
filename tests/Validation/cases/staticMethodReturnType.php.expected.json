{
    "references": {
        "FooClass": [
            "./staticMethodReturnType.php"
        ]
    },
    "definitions": {
        "FooClass": {
            "fqn": "FooClass",
            "extends": [],
            "isMember": false,
            "roamed": false,
            "isStatic": false,
            "canBeInstantiated": true,
            "symbolInformation": {
                "name": "FooClass",
                "kind": 5,
                "location": {
                    "uri": "./staticMethodReturnType.php"
                },
                "containerName": ""
            },
            "type": null,
            "declarationLine": "class FooClass {",
            "documentation": null,
            "signatureInformation": null
        },
        "FooClass::staticFoo()": {
            "fqn": "FooClass::staticFoo()",
            "extends": [],
            "isMember": true,
            "roamed": false,
            "isStatic": true,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "staticFoo",
                "kind": 6,
                "location": {
                    "uri": "./staticMethodReturnType.php"
                },
                "containerName": "FooClass"
            },
            "type__tostring": "\\FooClass",
            "type": {},
            "declarationLine": "public static function staticFoo(): FooClass {",
            "documentation": null,
            "signatureInformation": {
                "label": "()",
                "documentation": null,
                "parameters": []
            }
        },
        "FooClass::staticSelf()": {
            "fqn": "FooClass::staticSelf()",
            "extends": [],
            "isMember": true,
            "roamed": false,
            "isStatic": true,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "staticSelf",
                "kind": 6,
                "location": {
                    "uri": "./staticMethodReturnType.php"
                },
                "containerName": "FooClass"
            },
            "type__tostring": "\\FooClass",
            "type": {},
            "declarationLine": "public static function staticSelf(): self { }",
            "documentation": null,
            "signatureInformation": {
                "label": "()",
                "documentation": null,
                "parameters": []
            }
        },
        "FooClass->bar()": {
            "fqn": "FooClass->bar()",
            "extends": [],
            "isMember": true,
            "roamed": false,
            "isStatic": false,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "bar",
                "kind": 6,
                "location": {
                    "uri": "./staticMethodReturnType.php"
                },
                "containerName": "FooClass"
            },
            "type__tostring": "mixed",
            "type": {},
            "declarationLine": "public function bar() { }",
            "documentation": null,
            "signatureInformation": {
                "label": "()",
                "documentation": null,
                "parameters": []
            }
        }
    }
}